Committing Changes and Adding KeyboardDrive 1-28-22 Jack Tommaney

JackTWing opened the pull request:
Adds KeyboardDrive (yay!!!!!!!!!)

JackTWing committed:
Adds KeyboardDrive functionality, and test setup

The Keyboard Drive is now functional, with a basic testing setup.  It
does not have connectivity for changing the values in TestingDashboard
yet.  It also stops when turning.  Plans in the future include a more
overloaded constructor to fix this issue. 1-28-22

JackTWing committed:
Adds TestingDashboard registry for KeyboardDrive

Adds the registry for KeyboardDrive.java 1-28-22 Jack Tommaney

zzberries commented on src/main/java/frc/robot/OI.java:
```diff
@@ -8,9 +8,13 @@
 import frc.robot.input.AttackThree;
 import frc.robot.input.ButtonBox;
+import frc.robot.input.KeyboardBox;
+import frc.robot.input.KeyboardBox;
```
Please remove double import

zzberries commented on src/main/java/frc/robot/OI.java:
```diff
@@ -59,6 +67,14 @@ public OI() {
+    // Now Mapping Commands to Keyboard Box
+    ////////////////////////////////////////////////////
+
+    keyboardBox.getKForward().whileHeld(new KeyboardDrive(0.75, 0));
```
Perhaps we can give the constants names (defined in KeyboardDrive) and pass in the named constants

zzberries commented on src/main/java/frc/robot/commands/Drive/KeyboardDrive.java:
```diff
@@ -0,0 +1,75 @@
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.OI;
+import frc.robot.input.KeyboardBox;
+//import frc.robot.TestingDashboard;
```
Please remove the commented-out import

JackTWing replied:
I think we need it though?

snowpuppy replied:
We don't. If it's commented out, it's not required for the operation of the software. :)

zzberries commented on src/main/java/frc/robot/commands/Drive/KeyboardDrive.java:
```diff
@@ -0,0 +1,75 @@
+public class KeyboardDrive extends CommandBase {
+  /** Creates a new KeyboardDrive. */
+  private final Drive m_drive;
+  private static OI oi;
```
Suggestion: change oi to m_oi (and a similar thing with speed & rotation)

zzberries commented on src/main/java/frc/robot/input/KeyboardBox.java:
```diff
@@ -0,0 +1,209 @@
+	private JoystickButton IntakeIn;
+	private JoystickButton IntakeOut;
+	private JoystickButton Hopper;
+	//private JoystickButton CPDeploy;
```
Please remove commented-out code

zzberries commented on src/main/java/frc/robot/input/KeyboardBox.java:
```diff
@@ -0,0 +1,209 @@
+	 * 
+	 * @return Color Panel Deploy button
+	 */
+//	public JoystickButton getCPDeploy() {
```
Please remove commented-out methods

JackTWing committed:
Adding speed squaring for RaceDrive and ArcadeDrive

Adds speed scaling for the two drive methods mentioned above, as well as
adding the Intake subsystem base on 2-11-22  --  Jack Tommaney

JackTWing committed:
Merge remote-tracking branch 'upstream/master'

JackTWing committed:
Merge remote-tracking branch 'upstream/master'

snowpuppy commented:
This pull request has been resolved. The commits were merged into the main repository.
